### PR TITLE
Major revamp

### PR DESCRIPTION
**Pull request by joax**
Major Revamp to the Version

 * Javascript injection into separate file
 * Fixed unread notifications (div changed)
 * Moved notifications to Electron notifications from HTML5
 * Fixed issue with Configuration flags (always true)
 * Configuration changes write to Configuration File
 * Added Frameless Mode (and config on Tray Menu)
 * Frameless Mode: Added Grab areas
 * Frameless Mode: Added close button to child windows
 * TO-DO on child window rather than browser

Tested on Ubuntu 20.04 LTS. Needs testing on MacOS and Windows.

**Needs version bump**

### CODE REVIEW

File: src/controller/mail-window-controller.js [hunk 1]
```diff
@@ -57,35 +57,82 @@ class MailWindowController {
...
         // insert styles
         this.win.webContents.on('dom-ready', () => {
+            console.log('Insert CSS')
             this.win.webContents.insertCSS(CssInjector.main)
-            if (!showWindowFrame) this.win.webContents.insertCSS(CssInjector.noFrame)
-
-            this.addUnreadNumberObserver()
-
-            this.win.show()
+            let that = this
```
Comment (joax) [inline src/controller/mail-window-controller.js:69]: This needs some love. Old ways must be cleaned on future release.

File: src/controller/mail-window-controller.js [hunk 2]
```diff
@@ -57,35 +57,82 @@ class MailWindowController {
...
+        this.win.webContents.on('did-create-window', (childWindow) => {
+            // For example...
+            console.log('Child Window Detected..')
+            
+            // insert styles
+            childWindow.webContents.on('dom-ready', () => {
+                console.log('Insert CSS')
+                childWindow.webContents.insertCSS(CssInjector.main)
+
+                let that = this
```
Comment (joax) [inline src/controller/mail-window-controller.js:104]: Old ways to be cleaned.

**Comment by joax** [comment]
I have been using this PR for quite a while now and it works very well, some of the features are much needed (adding close buttons to the windows created on border-less mode). I can retract the PR if that cleans the repo, lmk.

**Comment by julian-alarcon** [comment]
I merged a PR to add missing spellcheck functionality, but it seems to be in conflict, can you review it? The changes was not so big @joax

**Review by julian-alarcon — CHANGES_REQUESTED** [review]
Also:
* I'm not sure if one intention of this PR is to provide this feature https://github.com/julian-alarcon/prospect-mail/issues/29 , I tested it and it didn't worked, notifications are still inside the window, not in the OS level. Could it be that something that is missing?
* I check your change to add deeplink for the ToDo, but it seems to still not opening a new Window.
* The unread  seems to be still broken, I see 3 dots instead of the tray icon. Can you show me an screenshot of how it looks at your end? (I'm using Ubuntu 20.04)

BTW Frameless feature work as intended!

**Comment by joax** [comment]
Merged. Detected one issue when borderless mode, the auxiliary windows open with border/window title.

**Comment by andrea-nisti** [comment]
Hello, notifications at OS level is a very appreciated feature. Any news about that? Thanks and awesome work!

**Review by julian-alarcon — DISMISSED** [review]

**Pull request closed without merging**